### DISCUSSION TITLE
Integration with Home Assistant

### DISCUSSION THREAD

**Discussion by luvxinh** [PJanisio/ewelinkApiPhp#18] [Ideas]
Hi.

I'm newbie in terms of coding but have been using Home Assistant for a long time now. In fact, the reason why I come across this repo is due to searching for a home assistannt integration.

So long story short, Home Assistant has this integration that will read the data from the ewelink devices. The problem is the data would not update if the device is not being "actively probe" by the ewelink app.

From your repo, I understood that you can probe the device.

I'm wondering if that "probe" part can be included as part of Home assistant? That would be great.

Thank you for your time

**Comment by PJanisio** [comment]
You are right, I spent a lot of time solving this riddle, on how to get always refreshed parameters.

The solution is using websocket connection and sending update command for the parameters, than wait like 1-3 seconds for update. That is why i made a dedicated function for it: **forceWakeUp** :

```php

            $forceWakeUpRes = $devs->forceWakeUp($devIdent);

            if ($forceWakeUpRes) {
                $allLiveParams = $devs->getAllDeviceParamLive($devIdent);
                echo '<h1>Get All Device Parameters Live After Force Wake Up</h1>';
                echo '<pre>' . print_r($allLiveParams, true) . '</pre>';
            }
```

Im not sure if I would have time to port it to HA, but you can also take this comment and paste the solution on HA github repo, so the owner can modify his integration.

**Reply by luvxinh** [reply]
Thanks. I've already linked your comment to the other thread ...

Just out of curiosity, we cannot just simply create a function (in Home Assistant) with your code and call that function every x seconds, right?

**Reply by PJanisio** [reply]
Im not so deeply in HA code, but HA components are written in Python and my code is written in PHP, so it will not work out of the box. It has to be ported and adjusted to what exactly HA core needs. I would really like to go deeply into HA components but I have other project to work on, so for sure until end of Aug - no time for it ,can you please share in which repo you linked this topic? Maybe in a free time I can look at it.

**Reply by luvxinh** [reply]
Hi.

I've posted it here.

It seems like he already has the answer, but it's still not working for me :D 

https://github.com/AlexxIT/SonoffLAN/issues/1425#issuecomment-2244969365

**Reply by PJanisio** [reply]
I see that it has been working for you, reading your last comment. Glad to hear. Im closing the discussion than. Take care!
